src/store/index.ts

```
@@ -37,62 +37,80 @@ const store = new Vuex.Store({
   },
 
   actions: {
+    // Should only get called once when the app is created
+    initProvider: async ({ dispatch }) => {
+      // If there's already a window.provider we don't need to do any additional setup
+      if (window.provider || window.provider != null) {
+        return;
+      }
+      const localProviderName = providers.getLocalProviderName();
+      if (!localProviderName || localProviderName === '') {
+        return;
+      }
+      dispatch('setProvider', { providerName: localProviderName });
+    },
+    // Called to initialize the app after a provider is set
     init: async ({ commit, dispatch }) => {
+      if (!window.provider || window.provider == null) {
+        return console.error('provider not set!');
```

Hmm, I wonder if we want to throw an alert error to at least notify the user

---

Hmm.. Wouldn't the users just hit the connect button?